.github/workflows/e2e-ios.yml

```diff
@@ -0,0 +1,39 @@
+name: E2E iOS
+
+on:
+  pull_request:
+    paths:
+      - packages/mobile/**
+
+jobs:
+  detox:
+    timeout-minutes: 10
+    runs-on: self-hosted
```

Comment:
This is arbitrary name we've given right? Perhaps there could be something more descriptive like self-hosted-macos-m2 ?

Reply:
It's all about labels, I'll be more explicit